sentry/src/main/java/io/sentry/SentryBaseEvent.java

```diff
@@ -186,7 +186,7 @@ public void setThrowable(final @Nullable Throwable throwable) {
   }
 
   public void setTags(@Nullable Map<String, String> tags) {
-    this.tags = tags;
+    this.tags = tags != null ? new HashMap<>(tags) : null;
```

Comment:
maybe worth adding a method to `CollectionUtils` that does all of this

Reply:
`CollectionUtils.newHashMap(@Nullable Map<K,V> map)` and rename `shallowCopy` to `CollectionUtils.newConcurrentHashMap(@Nullable Map<K,V> map)`?

Reply:
sounds good, maybe we add in the javadocs that this is pretty much a shallow copy as well, as its being used in a lot of cloning methods